### PR TITLE
(PA-4872) openssl 3 solaris 11 x86

### PR DESCRIPTION
**Pull request by cthorn42**

### CODE REVIEW

**Comment by joshcooper** [comment]
This needs rebasing due to merge conflicts.

**Comment by joshcooper** [comment]
ah it looks like there's a merge commit in the feature branch. Could you instead rebase these commits onto upstream/master like

```
git reset --hard 52bfb2b               
git rebase 099a647 --onto upstream/master
```

You'll get some conflicts which need to be resolved, then

```
git rebase --continue
git push -f origin HEAD
```

The reason for going through this trouble is merge commits in pull requests make it difficult to understand git history. So the rebase workflow is preferred.

File: configs/platforms/solaris-11-i386.rb [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
-    pl-cmake
-    pl-gcc-i386
-    pl-pkg-config
-  )
+   packages = %w(
+     pl-binutils-i386
+     pl-cmake
+     pl-gcc-i386
+     pl-pkg-config
+   )
```
Comment (joshcooper) [inline configs/platforms/solaris-11-i386.rb:10]: nit extra white space makes it look like there are changes, but only `gmake` is new

File: configs/projects/_shared-agent-settings.rb [hunk 2]
```diff
@@ -156,7 +156,7 @@
   proj.setting(:ldflags, "-L#{proj.libdir} -Wl,-rpath=#{proj.libdir},-z,relro,-z,now")
 end
 
-if ruby_version_x == "3" && !platform.is_solaris?
+if ruby_version_x == "3" #&& !platform.name =~ /solaris-11-sparc|solaris-10/
```
Comment (joshcooper) [inline configs/projects/_shared-agent-settings.rb:159]:
```suggestion
if ruby_version_x == "3"
```

**Comment by cthorn42** [comment]
Thanks for the help Josh, greatly appreciated.

File: configs/platforms/solaris-11-i386.rb [hunk 3]
```diff
@@ -40,7 +41,11 @@
 basedir=default" > /var/tmp/vanagon-noask;
   echo "mirror=https://artifactory.delivery.puppetlabs.net/artifactory/generic__remote_opencsw_mirror/testing" > /var/tmp/vanagon-pkgutil.conf;
   pkgadd -n -a /var/tmp/vanagon-noask -d http://get.opencsw.org/now all
+<<<<<<< HEAD
   /opt/csw/bin/pkgutil --config=/var/tmp/vanagon-pkgutil.conf -y -i libffi_dev autoconf gcc4core || exit 1;
+=======
+  /opt/csw/bin/pkgutil --config=/var/tmp/vanagon-pkgutil.conf -y -i libffi_dev autoconf gcc4core|| exit 1;
+>>>>>>> (PA-4872) OpenSSL 3 work for Solaris-11-x86-64
```
Comment (joshcooper) [inline configs/platforms/solaris-11-i386.rb:48]: ah looks like this conflict still needs resolving, note the `gcc4core ||` vs `gcc4core||` difference

**Review by joshcooper — COMMENTED** [review]
Looks good. If you haven't already, could you vanagon-generic-build these project & platform combinations:

```
❯ bx rake vanagon:component_diff -- -P agent-runtime-7.x -p all --from 46c127e --to HEAD  | grep 'Platform name'
Platform name: solaris-11-i386

❯ bx rake vanagon:component_diff -- -P agent-runtime-main -p all --from 46c127e --to HEAD  | grep 'Platform name' | grep -v solaris-10
Platform name: solaris-11-sparc
Platform name: solaris-11-i386
```

We can ignore solaris 10 on `agent-runtime-main`

**Comment by cthorn42** [comment]
Here is the solaris-11-i386 for agent-runtime-7.x: https://jenkins-platform.delivery.puppetlabs.net/view/vanagon-generic-builder/job/platform_vanagon-generic-builder_vanagon-packaging_generic-builder/2118/BUILD_TARGET=solaris-11-i386,SLAVE_LABEL=k8s-worker/
and here is the agent-runtime-main job: https://jenkins-platform.delivery.puppetlabs.net/view/vanagon-generic-builder/job/platform_vanagon-generic-builder_vanagon-packaging_generic-builder/2116/
The i386 cell failed with the rubygems-ffi failure that is ticketed, and the sparc cell failed. I guess that should be suspected since we haven't landed any changes for that yet?

**Review by joshcooper — APPROVED** [review]
Thanks for running those. I'm not concerned about the solaris 11 sparc failures in `agent-runtime-main`

**Pull request merged**